mlflow/protos/service.proto

```diff
@@ -934,11 +934,23 @@ message SearchRuns {
 
   optional string page_token = 7;
 
+  // These metrics, params and tags whitelist limit the fields returned by the search runs query
```

Comment:
not sure if you saw the [RFC for pagination](https://docs.google.com/document/d/1Ygti9oyMbi4DJNbaM0D49nNIKb8BLXwGYQ7WmrOKJ3s/edit?usp=sharing) we talked about in #2084 

The proposed mechanism is to use a single `column` field in `SearchRuns` proto. This would be similar to `order_by` field which is a comma separated list of columns for result ordering. 

Using `columns` field -- User would specify expected columns like `metrics.accuracy, metrics.mse, params.layers, tags.project`. Backend will parse this using the same logic uses for column names `filter` and `order_by` and return only these columns from the runs returned.

It would make sense to also return columns referred in `filter` and `order_by` query to handle pagination correctly.